src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// Taken from https://github.com/dotnet/cli/blob/8f7b58dd665f60300221a4eb5910a691749396aa/src/Microsoft.DotNet.MSBuildSdkResolver/Interop.NETStandard.cs
+
+#if NETSTANDARD2_0
+using System.Diagnostics;
+
+namespace Microsoft.Build.Locator
+{
+
+    using System;
+    using System.IO;
+    using System.Runtime.InteropServices;
+    using System.Text;
+
+    internal static class DotNetSdkLocationHelper
+    {
+        public static VisualStudioInstance GetInstance(string workingDir)
+        {
+            string dotNetSdkPath = hostfxr_resolve_sdk(workingDir);
+
+            if (String.IsNullOrWhiteSpace(dotNetSdkPath))
+            {
+                return null;
+            }
+
+            if (!File.Exists(Path.Combine(dotNetSdkPath, "Microsoft.Build.dll")))
+            {
+                return null;
+            }
+
+            FileInfo dotnetAssemblyPath = new FileInfo(Path.Combine(dotNetSdkPath, "dotnet.dll"));
+
+            if (!dotnetAssemblyPath.Exists)
+            {
+                return null;
+            }
+
+            FileVersionInfo fileVersionInfo = FileVersionInfo.GetVersionInfo(dotnetAssemblyPath.FullName);
+
+            return new VisualStudioInstance(
+                name: ".NET Core SDK",
+                path: dotNetSdkPath,
+                version: new Version(fileVersionInfo.FileMajorPart, fileVersionInfo.FileMinorPart, fileVersionInfo.FileBuildPart),
+                discoveryType: DiscoveryType.DotNetSdk);
+        }
+
+        private static string hostfxr_resolve_sdk(string workingDir)
+        {
+            Process process = Process.GetCurrentProcess();
+
+            string dotNetExePath = process?.MainModule?.FileName;
+
+            if (String.IsNullOrWhiteSpace(dotNetExePath))
+            {
+                return null;
```

Comment:
Should we throw here or provide the user with some explanation for why we couldn't locate MSBuild? If it fails, it feels like this is going to be a head-scratcher. There are a lot of failure conditions here that we could provide (somehow) more information to the user about.

Reply:
This might be a way of detecting standalone app vs shared app.  If the main module isn't `dotnet.exe`, it must be standalone right?  Maybe throw here saying to not use MSBuildLocator?

Reply:
I'm not sure what the right thing to do would be. @livarcocc, do you have any ideas for what we could do for standalone apps?

Reply:
I wouldn't expect standalone apps to fail here, but instead fail to find any sdks below.